src/randomization.ts

```diff
@@ -159,12 +159,12 @@ function fixOrderForConstraints(
  * @returns {null|[]} The randomized order. Null if randomization
  * failed, in which case an error will have been logged to the console
  */
-function randomizePrivate(
-    original_stimuli: Array<StringToUnknownMapping>,
+function randomizePrivate<TItem extends ItemType>(
+    original_stimuli: Array<TItem>,
     constraints: ConstraintMapping,
     nth_try: number,
     max_tries: number
-) : Array<StringToUnknownMapping> | null {
+) : Array<TItem> | null {
```

Comment:
Here the generic really shines, by returning the generic, TS knows it's getting the exact same type back instead of the less-usable `Record<string, unknown>`.